plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergMinioConnectorTest.java

```diff
@@ -0,0 +1,234 @@
+     * <p>
+     * Limiting dataset to 1500 rows in dedicated test methods for MinIO solves this issue:
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnCtasPartitionComplex()}
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnInsertPartitionComplex()}
+     * <p>
+     * 2. partitioning -> 'truncate(comment, 1)', 'orderstatus', 'bucket(comment, 2)'
+     * Test environment causes HMS to consume more time during createTable operation.
+     * This leads to read timed out and retry operation ended with TableAlreadyExists
+     * (HMS managed to end operation on it's side). As such behaviour wasn't observed
+     * in normal circumstances, test was adopted to avoid it.
+     * <p>
+     * Limiting dataset to 300 rows in dedicated test methods for MinIO solves this issue:
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnCtasPartitionWithSameColumnMultipleTimes()}
+     * - {@link BaseIcebergMinioConnectorTest#testRepartitionDataOnInsertPartitionWithSameColumnMultipleTimes()}
+     */
+    @Override
+    @DataProvider
+    public Object[][] repartitioningDataProvider()
+    {
+        Session obeyConnectorPartitioning = Session.builder(getSession())
+                .setSystemProperty(PREFERRED_WRITE_PARTITIONING_MIN_NUMBER_OF_PARTITIONS, "1")
+                .build();
+        return new Object[][] {
+                // identity partitioning column
+                {obeyConnectorPartitioning, "'orderstatus'", 3},
+                // bucketing
+                {getSession(), "'bucket(custkey, 13)'", 13},
+                // varchar-based
+                {getSession(), "'truncate(comment, 1)'", 35}
+        };
+    }
+
+    @Test
+    @Override
+    public void testShowCreateTable()
+    {
+        assertThat((String) computeScalar("SHOW CREATE TABLE region"))
+                .isEqualTo("" +
+                        "CREATE TABLE iceberg.tpch.region (\n" +
+                        "   regionkey bigint,\n" +
+                        "   name varchar,\n" +
+                        "   comment varchar\n" +
+                        ")\n" +
+                        "WITH (\n" +
+                        "   format = '" + format + "',\n" +
+                        "   location = 's3://" + bucketName + "/tpch/region'\n" +
+                        ")");
+    }
+
+    @Test
+    @Override
+    public void testRenameSchema()
+    {
+        // Overridden because error message is different from upper test method
+        String schemaName = getSession().getSchema().orElseThrow();
+        assertQueryFails(
+                format("ALTER SCHEMA %s RENAME TO %s", schemaName, schemaName + randomTableSuffix()),
+                "Hive metastore does not support renaming schemas");
+    }
+
+    @Test
+    @Override
+    public void testCreatePartitionedTableAs()
+    {
+        testCreatePartitionedTableAs("s3://" + bucketName + "/tpch/tmp_" + randomTableSuffix());
```

Comment:
instead of having to override each test that uses a table location, maybe we could have a method like `createNewTableLocation`.
This would makes the superclass simpler (no need to split test method into (1) entry point and (2) implementation), and reduce number of overrides

Reply:
Applied